src/controller.js

```diff
@@ -53,7 +53,13 @@ async function getAuth(req, res) {
       const user = await userModel.findOne({ email: authorizationData.userId }).lean();
       if (!user) {
         res.status(404).send();
-      } else if (authorizationData.password !== user.password) {
+      }
+      // Compares the provided password with the stored hashed password
+      const isPasswordMatch = await userModel
+        .findById(user._id)
+        .then((u) => u.comparePassword(authorizationData.password));
```

Comment:
Please change this from a promise pattern to an `async/await` pattern.

Also, `u` is a single-letter argument -- it should have a more descriptive name.